services/articles/articles_services_test.go

```diff
@@ -0,0 +1,97 @@
+package services
+
+import (
+	"testing"
+
+	"backend/mocks" // Import the generated mock package
+	"backend/pkg/models"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestArticleService_GetAllArticles(t *testing.T) {
```

Comment:
Better if you cover -ve test cases here for all func with Table Driven testing.
GetAllArticles, GetArticleByID & CreateArticle